eng/pipelines/templates/steps/cache-maven-repository.yml

```diff
@@ -0,0 +1,281 @@
+steps:
+- script: |
+    echo "##vso[task.setvariable variable=Maven.RepositoryPath;]%USERPROFILE%\.m2\repository"
+  condition: and(eq(variables['Agent.OS'], 'Windows_NT'), ne(variables['Build.Reason'], 'PullRequest'), eq(variables['EnableCache'], 'true'))
+  displayName: Detecting Maven repository on Windows
+- script: |
+    echo "##vso[task.setvariable variable=Maven.RepositoryPath;]$HOME/.m2/repository"
+  condition: and(ne(variables['Agent.OS'], 'Windows_NT'), ne(variables['Build.Reason'], 'PullRequest'), eq(variables['EnableCache'], 'true'))
+  displayName: Detecting Maven repository on Linux and macOS
+- task: CacheBeta@0
+  inputs:
+    key: |
+      JobName-$(Agent.JobName)
+      CacheSalt-$(CacheSalt)
+      EnableJetty-$(EnableJetty)
+      $(Build.SourcesDirectory)/pom.client.xml
```

Comment:
ugh... you have to explicitly list all the files?

Reply:
Currently - yes, there is a top feature request for pipeline caching to support globbing here. For the time being I thought I would centralize this horror of a fingerprint list in a separate template that we can just reference in ;) Fingerprinting all these files takes something like 2 seconds (tops). Hopefully by the time we take a second pass at this we can just replace it with a glob!

Reply:
It should be pretty easy to replace it with a glob in a script block and set it as a variable we use in this block.